🌱 Make MachinesByCreationTimestamp private to machine collections (#6111)
kubernetes-sigs/cluster-api: docs/book/src/developer/providers/v1.1-to-v1.2.md

### Deprecation

> * `util.MachinesByCreationTimestamp` has been deprecated, as we are now using `util/collections.machinesByCreationTimestamp`

Reviewer: ```suggestion
* `util.MachinesByCreationTimestamp` has been deprecated and will be removed in a future release.
```
nit. I would write it like this. I think from an API consumer perspective it doesn't matter that we are still using a copy internally.

Author: Done.
